force-app/main/default/classes/RD2_ValidationService.cls

```diff
@@ -633,6 +621,7 @@ public with sharing class RD2_ValidationService {
      * @description Validate the NumberOfPlannedInstallments field value for Fixed Length and
      * Open Recurring Donations
      * @param rd The recurring donation to validate
+     * @param errorCollection Error wrapper which contains all error related to the RD
      */
     private void validateFixedLength(npe03__Recurring_Donation__c rd, ErrorRecord errorCollection) {
         RD2_RecurringDonation rdRecord = new RD2_RecurringDonation(rd);
```

Comment:
I think we can pass recurringDonation here and eliminate new RD2_RecurringDonation(rd);

Reply:
Yup, missed that one for some reason.